apps/cowswap-frontend/src/common/hooks/usePendingApprovalModal.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+
+import { getIsMetaMask, getWeb3ReactConnection, injectedConnection } from '@cowprotocol/wallet'
+import { useWeb3React } from '@web3-react/core'
+
+import { useModalState } from './useModalState'
+
+import { MediumAndUp, useMediaQuery } from '../../legacy/hooks/useMediaQuery'
+import { ConfirmationPendingContent } from '../pure/ConfirmationPendingContent'
+import { MetamaskApproveBanner } from '../pure/MetamaskApproveBanner'
```

Comment:
nit: Probably not worth it, but consider lazy loading if this component is only used in a specific wallet

Reply:
It's too small, don't worth it